### PR TITLE
Issue #6207: Add XPath IT Regression test for ParameterNumberCheck

### PR DESCRIPTION
**Pull request by mahfouz72**
issue #6207 [(Documentation)](https://checkstyle.org/checks/sizes/parameternumber.html#ParameterNumber)
1 test for default config, 1 for methods check, 1 for ignore override

### CODE REVIEW

**Review by romani — APPROVED** [review]
ok to merge

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNumberTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    }
+
+    @Test
+    public void testMethods() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionParameterNumberMethods.java"));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(ParameterNumberCheck.class);
+        moduleConfig.addAttribute("max", "10");
```
Comment (romani) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNumberTest.java:67]: CI is not happy of this usage
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/23705/workflows/ec082a1a-878e-429b-a0bd-a602f97a6c95/jobs/510223/artifacts
please recheck how others do not use this deprecation.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNumberTest.java (hunk 1 above)
Comment (mahfouz72) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNumberTest.java]: @romani done. used addProperty instead

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**